test/test_foreach.py

```diff
@@ -85,26 +85,39 @@ def _test_unary_op(self, device, dtype, foreach_op, foreach_op_, torch_op):
-            value = 2
 
-            # Mimics cuda kernel dtype flow.  With fp16/bf16 input, runs in fp32 and casts output back to fp16/bf16.
-            control_dtype = torch.float32 if (self.device_type == 'cuda' and
-                                              (dtype is torch.float16 or dtype is torch.bfloat16)) else dtype
-            expected = [torch_op(tensors[i].to(dtype=control_dtype),
-                                 tensors1[i].to(dtype=control_dtype),
-                                 tensors2[i].to(dtype=control_dtype), value=value).to(dtype=dtype) for i in range(N)]
+            for values in [value, [value for _ in range(N)]]:
+                # Mimics cuda kernel dtype flow.  With fp16/bf16 input, runs in fp32 and casts output back to fp16/bf16.
+                control_dtype = torch.float32 if (self.device_type == 'cuda' and 
+                                                  (dtype is torch.float16 or dtype is torch.bfloat16)) else dtype
+                expected = [torch_op(tensors[i].to(dtype=control_dtype),
+                                     tensors1[i].to(dtype=control_dtype),
+                                     tensors2[i].to(dtype=control_dtype),
+                                     value=value).to(dtype=dtype) for i in range(N)]
+
+                res = foreach_op(tensors, tensors1, tensors2, values)
+                foreach_op_(tensors, tensors1, tensors2, values)
+                self.assertEqual(res, tensors)
+                if (dtype is torch.float16 or dtype is torch.bfloat16) and TEST_WITH_ROCM:
+                    self.assertEqual(tensors, expected, atol=1.e-3, rtol=self.dtype_precisions[dtype][0])
+                else:
+                    self.assertEqual(tensors, expected)
 
-            res = foreach_op(tensors, tensors1, tensors2, value)
-            foreach_op_(tensors, tensors1, tensors2, value)
-            self.assertEqual(res, tensors)
-            if (dtype is torch.float16 or dtype is torch.bfloat16) and TEST_WITH_ROCM:
-                self.assertEqual(tensors, expected, atol=1.e-3, rtol=self.dtype_precisions[dtype][0])
-            else:
-                self.assertEqual(tensors, expected)
+        # test error cases
+        tensors = self._get_test_data(device, dtype, N)
+        tensors1 = self._get_test_data(device, dtype, N)
+        tensors2 = self._get_test_data(device, dtype, N)
+
+        for op in [torch._foreach_addcmul, torch._foreach_addcmul_, torch._foreach_addcdiv, torch._foreach_addcdiv_]:
+            with self.assertRaisesRegex(RuntimeError, "Tensor list must have same number of elements as scalar list."):
+                torch._foreach_addcmul(tensors, tensors1, tensors2, [value for _ in range(N + 1)])
+            with self.assertRaisesRegex(RuntimeError, "Tensor list must have same number of elements as scalar list."):
+                torch._foreach_addcmul(tensors, tensors1, tensors2, [value for _ in range(N - 1)])
```

Comment:
This test isn't comprehensive: to be comprehensive, we would also test what happens if some of `{tensors, tensors1, tensors2}` don't have the same sizes.